From 41c03ca2298848be052bcd2c5967d090eff23164 Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Tue, 1 Oct 2024 15:01:44 +0200
Subject: [PATCH 1550/1695] module/scmi: fix scmi_api->respond size

outlined by coverity

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c3332f3d42c8f0e04138160379c5c57b11297c24 module/scmi: fix scmi_api->respond size]
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
Change-Id: Ieddd9af12901c97334e9f4dadc0daa77a46be7c6
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/408743
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Simeon MARIJON <simeon.marijon@st.com>
Reviewed-by: Simeon MARIJON <simeon.marijon@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 .../module/scmi_voltage_domain/src/mod_scmi_voltage_domain.c    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/lib/scmi-server/SCP-firmware/module/scmi_voltage_domain/src/mod_scmi_voltage_domain.c b/core/lib/scmi-server/SCP-firmware/module/scmi_voltage_domain/src/mod_scmi_voltage_domain.c
index 0c65e4b78..f38d35040 100644
--- a/core/lib/scmi-server/SCP-firmware/module/scmi_voltage_domain/src/mod_scmi_voltage_domain.c
+++ b/core/lib/scmi-server/SCP-firmware/module/scmi_voltage_domain/src/mod_scmi_voltage_domain.c
@@ -637,7 +637,7 @@ exit:
         respond_status = scmi_api->respond(service_id, NULL, payload_size);
     } else {
         respond_status = scmi_api->respond(
-            service_id, &outmsg.status, sizeof(&outmsg.status));
+            service_id, &outmsg.status, sizeof(outmsg.status));
     }
 
     if (respond_status != FWK_SUCCESS) {
-- 
2.47.2

